### DISCUSSION TITLE
How to make an environment similar to the live competition

### DISCUSSION THREAD

**Discussion by a4rcvv** [yasserfarouk/scml#103] [one-shot-qa]
I’m using this code to evaluate my agent. (Based on [Running a One-Shot tournament](http://www.yasserm.com/scml/scml2020docs/tutorials/01.run_scml2020.html#running-a-one-shot-tournament))

```python
if __name__ == '__main__':
    types = [
        AgentFoo,
        AgentBar,
    ]
    results = anac2022_oneshot(
        competitors=types,
        n_configs=50,
        n_runs_per_world=1,
        n_steps=100,
        print_exceptions=True
    )
    results = shorten_names(results)
    print(results.score_stats)
    print(results.total_scores)
```

I use the agents in `scml_agents` as the competitors and the score of my agent is 1.0 or higher, so I think my agent works well in this code. However, in the live competition, the score is about 6000. I think it is too weak compared to the result in the local environment. (It is based on the assumption that a displayed score in the Online Tournament is 10,000 times as same as that in the local environment)

I’m suspecting (but not sure) that my agent doesn’t make any contracts in the live competition, because the score of it is similar to the score of the agent which responds END_NEGOTIATION immediately.

So, I have to identify what is happening in the live competition. Is there better way to reproduce the environment of it and the official competition?

Here is the versions of Python and libraries I’m using.

- Python 3.8.13
- scml 0.5.5
- negmas 0.9.5

**Comment by yasserfarouk** [comment]
Unfortunately, the only way to reproduce the online environment is to actually have access to other competitor agents. Nevertheless, you are right to suspect that something is not right.
I can help you a little if I know which agent is yours. At least I can tell you whether or not it is making any contracts.

**Reply by a4rcvv** [reply]
Thank you very much for your reply and help. My agent is "AgentAM", whose ID is 20143.

**Reply by yasserfarouk** [reply]
I checked the recent tournaments (few days ago as it seems you are changing the active agent). It seems that AgentAM does get some agreements but few of them.

**Reply by a4rcvv** [reply]
Thank you. I changed the active agent to an old one to check the behavior in the live competition and it seems that it works well, so I'm going to compare `AgentAM` with it.
